federation/pkg/federation-controller/service/service_controller.go

```diff
@@ -614,6 +614,23 @@ func (sc *ServiceController) reconcileService(serviceKey types.NamespacedName) (
 			return statusError, err
 		}
 
+		// Copy ingresses of unreachable clusters for comparison and do not delete those ingresses
+		unreadyClusters, err := sc.serviceFederatedInformer.GetUnreadyClusters()
+		if err != nil {
+			glog.Errorf("Failed to get unready cluster list: %v", err)
+			return statusError, err
+		}
+
+		for _, cluster := range unreadyClusters {
+			region := cluster.Status.Region
+			zone := cluster.Status.Zones[0]
+			clusterIngress := getClusterServiceIngresses(existingServiceIngress, region, zone, cluster.Name)
+			if clusterIngress != nil {
+				getOrCreateClusterServiceIngresses(newServiceIngress, region, zone, cluster.Name)
```

Comment:
yes @quinton-hoole it compiles. in fact, if you assign the return code to _ as i am not using it here, it complains. i agree with you that logic is weird, will replace by better function name/code.